### DISCUSSION TITLE
Ability to install a single hook (like lodash.memoize, or lodash.<insert lodash function here>

### DISCUSSION THREAD

**Discussion by wadehammes** [juliencrn/usehooks-ts#605] [Ideas]
We would love the ability to install via package manager a single instance of a hook, rather than the whole library. We don't want to give access to all the hooks here to our dev team, in a perfect world, we could just install the hook we want to use.

i.e. `npm install usehooks-ts/useSessionStorage`

**Comment by cafesanu** [comment]
@juliencrn wondering if there is been a talk about this idea?. Would love the ability to just install the hooks that I actually need only. 

Two reasons we would love this:
  1. Of course, we want our bundle to not include stuff we don't use, but even assuming this library is tree-shakable, reason `#2` is way more important for us
  2. I's not about this being tree shakable as much as not exposing hooks we don't want to use. We've vetted some of the hooks as appropriate for our repo while we don't want to (on purpose) use other ones in here. Installing the whole library leaves us exposed to not follow our own standards as any dev can mistakenly use a hook that we don't want to use.

Also, this is the most voted discussion 😉

Thanks!!
